### PR TITLE
Bug 1793093: [release-4.3] Fix machine config apply on upgrade

### PR DESCRIPTION
**Pull request by mtnbikenc**
* Use local pull-secret for pulling mcd image
* Use node desiredConfig to apply machine configuration that MCD expects
* Use proxy if defined

Backports #12069

### CODE REVIEW

File: roles/openshift_node/tasks/apply_machine_config.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
+---
+- name: Create temp directory
+  tempfile:
+    state: directory
+  register: temp_dir
+
+- name: Get cluster pull-secret
```
Comment (vrutkovs) [inline roles/openshift_node/tasks/apply_machine_config.yml]: Isn't pull secret already stored on the node, somewhere in `/var/lib/kubelet`?

Comment (mtnbikenc) [reply roles/openshift_node/tasks/apply_machine_config.yml]: Confirmed `/var/lib/kubelet/config.json`.
I'll update to use that instead.

**Pull request merged**